src/platform/i386/chal_pgtbl.c

```diff
@@ -0,0 +1,533 @@
+/**
+ * Copyright 2018 by Runyu Pan, panrunyu@gwu.edu
+ *
+ * Redistribution of this file is permitted under the GNU General
+ * Public License v2.
+ */
+#include "shared/cos_types.h"
+#include "captbl.h"
+#include "pgtbl.h"
+#include "cap_ops.h"
+#include "liveness_tbl.h"
+#include "retype_tbl.h"
+#include "chal/chal_prot.h"
+#include "chal_pgtbl.h"
+
+/* These functions do flag operations */
+unsigned long
+chal_pgtbl_flag_add(unsigned long input, pgtbl_flags_t flags)
```

Comment:
Just wonder if they should instead be MACROS. Or should you just `inline` them in the headerfile.